src/main/scala/shine/cuda/ast/Printer.scala

```diff
@@ -0,0 +1,80 @@
+package shine.cuda.ast
+
+import arithexpr.arithmetic.ArithExpr
+import shine.C.AST._
+import shine.DPIA.Types.AddressSpace
+import shine.{OpenCL, cuda}
+import shine.OpenCL.AST.KernelDecl
+import shine.cuda.BuiltInAttribute
+
+object Printer {
+  def apply(n: Node): String = (new Printer).printNode(n)
+}
+
+class Printer extends shine.OpenCL.AST.Printer {
+  override def printStmt(s: Stmt): Unit = s match {
+    case _: SynchronizeThreads => print("__syncthreads();")
+    case _: SynchronizeWarp => print("__syncwarp();")
+    case _ => super.printStmt(s)
+  }
+
+  override def toString(e: ArithExpr): String = e match {
+    case of: BuiltInAttribute => of.toString
+
+    case _ => super.toString(e)
+  }
+
+  override def printKernelDecl(k: KernelDecl): Unit = {
+    //TODO
```

Comment:
what is TODO?

Reply:
At the moment for every generated CUDA-Kernel the include-statement for the "<mma.h>"-header is printed. This should be removed and replaced by an include directives when use the tensor core primitives (when include directives are avaiable).